library/std/src/os/mod.rs

```diff
@@ -78,18 +78,18 @@ pub mod linux;
 #[cfg(not(all(
     doc,
     any(
-        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_arch = "wasm32", not(target_os = "wasi"), not(target_os = "wasi-preview2")),
```

Comment:
One possible idea for this would be to add `target_family = "wasi"`. I think there's already `target_family = "wasm"` and adding a second for `"wasi"` may be applicable for this (and the old) targets

Reply:
(also a reminder for the resolved conversation just beneath this that I think it would be good to either remove the `wasi` module here or duplicate it as `wasi_preview2` to avoid accidentally breaking it.